include/swift/ABI/Metadata.h

```diff
@@ -804,14 +804,39 @@ struct TargetVTableDescriptorHeader {
   /// entries occupy in instantiated class metadata.
   uint32_t VTableSize;
 
-  uint32_t getVTableOffset(const TargetClassMetadata<Runtime> *metadata) const {
-    const auto *description = metadata->getDescription();
-    if (description->hasResilientSuperclass())
-      return metadata->Superclass->getSizeInWords() + VTableOffset;
+  uint32_t getVTableOffset(const TargetClassDescriptor<Runtime> *description) const {
+    if (description->hasResilientSuperclass()) {
+      auto bounds = description->getMetadataBounds();
+      return (bounds.ImmediateMembersOffset / sizeof(StoredPointer)
+              + VTableOffset);
+    }
+
     return VTableOffset;
   }
 };
 
+template <typename Runtime>
+struct TargetMethodOverrideDescriptor {
+  /// The class containing the base method.
+  TargetRelativeIndirectablePointer<Runtime, TargetClassDescriptor<Runtime>> Class;
+
+  /// The base method.
+  TargetRelativeIndirectablePointer<Runtime, TargetMethodDescriptor<Runtime>> Method;
```

Comment:
Note that this is imposing an alignment restriction on method descriptor symbols.  I think that's okay, but it's something to document about them.

Reply:
Ok, good point.